docs: update react native sandboxing (#16394)
PostHog/posthog.com: contents/docs/error-tracking/upload-source-maps/react-native.mdx

<CalloutBox icon="IconInfo" title="CLI version requirement" type="action">

A minimum CLI version of [0.7.8](https://github.com/PostHog/posthog/releases/tag/posthog-cli%2Fv0.7.8) is required, but we recommend [keeping up with the latest CLI version](/docs/error-tracking/upload-source-maps/cli) to ensure you have all of error tracking's features.

</CalloutBox>

<CalloutBox icon="IconInfo" title="Git metadata on uploads" type="fyi">

> When you upload source maps from a local build or from CI, PostHog will attach git metadata (branch, commit, repo, remote URL) to the release so symbol sets link back to the exact commit in the Error Tracking UI. This works automatically for Android and iOS; no configuration is required.

Reviewer: do we mention this for all the other ways of uploading symbols? Is it important for customers? if not, i think this is just noisy in the UI

Author: removed

+/bin/sh `"$NODE_BINARY" --print "require('path').join(require('path').dirname(require.resolve('posthog-react-native')), '..', 'tooling', 'posthog-xcode.sh')"` `"$NODE_BINARY" --print "require('path').dirname(require.resolve('react-native/package.json')) + '/scripts/react-native-xcode.sh'"`
```

> You must also disable User Script Sandboxing for the upload script to work. In Xcode, go to Build Settings, search for **User Script Sandboxing** (`ENABLE_USER_SCRIPT_SANDBOXING`) and set it to **No**. This is required because the script needs to read `.git/` for release metadata and execute `posthog-cli`, which are not allowed under Xcode's script sandbox. If you are using the Expo plugin (step 1), this is handled automatically.

Reviewer 2: nit: I would start with "If you are not using the Expo plugin, " cause people can easily skip all the info that comes next if not aplicable

```suggestion
If you are not using the Expo plugin (step 1), you will need to also disable User Script Sandboxing for the upload script to work. In Xcode, go to Build Settings, search for **User Script Sandboxing** (`ENABLE_USER_SCRIPT_SANDBOXING`) and set it to **No**. This is required because the script needs to read `.git/` for release metadata and execute `posthog-cli`, which are not allowed under Xcode's script sandbox.
```

</Step>

<StepVerifySymbolSetsUpload />

</Steps>

> ## Plugin options

Reviewer: i think this should be a bit above, as part of the posthog-react-native/expo installation step instead, right now this is below the 'verify source maps upload' section which is the end of everything, feels weird

Example opting out of the sandbox change:

```json
{
  "expo": {
    "plugins": [
      ["posthog-react-native/expo", { "disableSandboxing": false }]
    ]
  }
}
```

<CalloutBox icon="IconInfo" title="When to opt out" type="fyi">

`disableSandboxing: false` is intended for organizations with compliance requirements that forbid disabling Xcode's user script sandbox. With the opt-out:

- iOS source map uploads continue to work — errors still symbolicate
- Git metadata will not be attached to iOS releases (the UI chip shown for Android rows will be missing on iOS rows)
- Your project must already have every build-phase script (yours, Expo's, CocoaPods-generated) declaring its inputs and outputs, otherwise the build will fail

This setting only affects build-time scripts on the build machine. Your shipped app's runtime security is unchanged.

</CalloutBox>

> ## Troubleshooting

Reviewer: should this be under https://c9c46085.posthog-preview.pages.dev/docs/error-tracking/troubleshooting instead? or is it very posthog react native specific?

Author: move generic part to that page and left one part here
